### PR TITLE
Docs: Documentation manual build output path correction

### PR DESCRIPTION
**Pull request by dottharun**
# Motivation

[Here](https://nixos.org/manual/nix/unstable/contributing/documentation#building-the-manual) in incremental building of the manual part,
the current documentation says output is in the `./outputs/out/` sub-directory, 
but the build command actually produces output at `./outputs/doc/` sub-directory

image

# Context

- The `--docdir` flag used to build the output seems to be a [standard directory variable](https://www.gnu.org/software/automake/manual/automake.html#Standard-Directory-Variables) of automake tools
- current build process of man-pages [Here](https://github.com/dottharun/nix/blob/4e2f11b6927cb0e4171b2758196f2310760617d2/doc/manual/local.mk#L177)
- This PR captures my learning of the Nix code and marks my first contribution to the codebase

# Priorities and Process

Add :+1: to [pull requests you find important](https://github.com/NixOS/nix/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

The Nix maintainer team uses a [GitHub project board](https://github.com/orgs/NixOS/projects/19) to [schedule and track reviews](https://github.com/NixOS/nix/tree/master/maintainers#project-board-protocol).

### CODE REVIEW

**Review by thufschmitt — APPROVED** [review]
Indeed, thanks :)

I don't merge right now as you've made this a draft. Let me know (or undraft) if that wasn't on purpose.

**Comment by dottharun** [comment]
@thufschmitt thanks for the speedy review, yes there's no need this to be draft

**Pull request merged**
